sqllogictest/src/substitution.rs

```diff
@@ -11,25 +11,44 @@ pub(crate) struct Substitution {
     test_dir: Arc<OnceLock<TempDir>>,
 }
 
+#[derive(thiserror::Error, Debug)]
+#[error("substitution failed: {0}")]
+pub(crate) struct SubstError(subst::Error);
+
+impl Substitution {
+    pub fn substitute(&self, input: &str, subst_env_vars: bool) -> Result<String, SubstError> {
+        if !subst_env_vars {
+            Ok(input
+                .replace("$__TEST_DIR__", &self.test_dir())
+                .replace("$__NOW__", &self.now()))
```

Comment:
I think we can also drop the support for them. Users can easily write something like `/tmp/$(uuidgen)` or `$(date +%s)` in shell script to achieve similar effects.